torch/jit/_trace.py

```diff
@@ -148,7 +148,7 @@ def clone_input(a):
             # TODO: figure out one liner to .clone() and set requires_grad
             v = (
                 a.detach()
-                .clone(memory_format=torch.preserve_format)
+                .clone(memory_format=None if a.is_mkldnn else torch.preserve_format)
```

Comment:
what is the purpose of this line? I don't actually know what `memory_format=None` does. Since we're converting the tensors to dense for comparison, why do we need this?

Reply:
For MKLDNN tensor, it doesn't have memory_format attribute, there has a check that the memory_format should be **None** when calling mkldnn_clone:https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/aten/src/ATen/native/mkldnn/TensorShape.cpp#L58-L62